eclipse-collections-api/src/main/java/org/eclipse/collections/api/list/ListIterable.java

```diff
@@ -279,7 +279,10 @@
      *
      * @see Collections#binarySearch(List, Object, Comparator)
      */
-    int binarySearch(T key, Comparator<? super T> comparator);
+    default int binarySearch(T key, Comparator<? super T> comparator)
+    {
+        return Collections.binarySearch((List<? extends T>) this, key, comparator);
+    }
 
     /**
      * Searches for the specified object using the binary search algorithm. The elements in this list must implement
```

Comment:
Note that this overload `binarySearch(T key)` already casts `this` and is already default. Not sure why we only did one of the two methods, but I'm guessing it had to do with complex generics.

Reply:
Interesting.  Good catch.